cosmos/dbt/graph.py

```diff
@@ -654,6 +654,8 @@ def load_via_custom_parser(self) -> None:
         for model_name, model in models:
             config = {item.split(":")[0]: item.split(":")[-1] for item in model.config.config_selectors}
             tags = [selector for selector in model.config.config_selectors if selector.startswith("tags:")]
+            # Remove the "tags:" prefix
+            tags = [tag.split(":")[-1] for tag in tags]
```

Comment:
We should not represent tags in different ways depending on the parsing mode. Either we remove the prefix for all  `LoadMode`s that set this configuration (not only `LoadMode.CUSTOM`, but also `LoadMode.DBT_MANIFEST`), or we remove for none of them.

By changing this here, I'm afraid we may be breaking the behaviour ahead in the selector, for instance:
https://github.com/astronomer/astronomer-cosmos/blob/main/cosmos/dbt/selector.py#L177

The prefix `"tag:" is currently being used in the `selector` module in at least four places via the `TAG_SELECTOR` constant: https://github.com/astronomer/astronomer-cosmos/blob/110fb0760cb1a21a2a21ca1a06370c44613ba085/cosmos/dbt/selector.py#L20C1-L20C13

Since the `select_nodes` is being used not only by `RenderMode.CUSTOM`, but also by `RenderMode.DBT_MANIFEST`, I believe this change will break the overall behaviour.

I'm surprised no unit tests broke.